### DISCUSSION TITLE
Geocentric altitude/az of the sun and moon.

### DISCUSSION THREAD

**Discussion by hidp123** [skyfielders/python-skyfield#845] [Q&A]
Normally, alt & az are topocentric. However, how can I get geocentric alt & az. The altitude at least, if not azimuth.

I ask mainly due to the following:
![image](https://user-images.githubusercontent.com/84683703/231354076-5c011bb8-0c41-4c4b-83c2-bbea4e12f8ee.png)

**Comment by brandon-rhodes** [comment]
The center of the Earth, the geocenter, does not have a horizon — so there would be no way to compute an angular altitude above that horizon, I don't think?

**Reply by hidp123** [reply]
Agree, it is definitely a strange one.

So far looking at other libraries I have found Swiss Ephemeris has this option, and cosinekitty at [astronomy-engine](https://github.com/cosinekitty/astronomy) managed to put together something for me in this regard using his library.
[Here](https://github.com/cosinekitty/astronomy/discussions/299#discussioncomment-5587004) (Swiss Ephemeris) and [here](https://github.com/cosinekitty/astronomy/discussions/299#discussioncomment-5598927) is my discussion with cosinekitty in regards to this.
I was hoping it can also be calculated via skyfield due to skyfield's high level of accuracy.

@brandon-rhodes
